lib/Sema/CSSimplify.cpp

```diff
@@ -2697,6 +2697,14 @@ ConstraintSystem::matchTypes(Type type1, Type type2, ConstraintKind kind,
     }
   }
 
+  if (type1->getOptionalObjectType() && type2->isAny()) {
+    auto *fix = AllowImplicitCoercionToAny::create(
+        *this, type1, type2, getConstraintLocator(locator));
+    recordFix(fix);
+    return matchTypes(type1->lookThroughAllOptionalTypes(), type2,
```

Comment:
This was giving me some trouble and still is - I am not sure if this is the right logic.